manifests/init.pp

```diff
@@ -115,104 +133,65 @@
-    enable_basic_auth            => false,
-    consumer_system_name_pattern => '.+',
-    adapter_module               => 'org.candlepin.katello.KatelloModule',
-    amq_enable                   => true,
-    amqp_keystore_password       => $::certs::candlepin::keystore_password,
-    amqp_truststore_password     => $::certs::candlepin::keystore_password,
-    amqp_keystore                => $::certs::candlepin::amqp_keystore,
-    amqp_truststore              => $::certs::candlepin::amqp_truststore,
-    qpid_ssl_cert                => $::certs::qpid::client_cert,
-    qpid_ssl_key                 => $::certs::qpid::client_key,
-    db_host                      => $candlepin_db_host,
-    db_port                      => $candlepin_db_port,
-    db_name                      => $candlepin_db_name,
-    db_user                      => $candlepin_db_user,
-    db_password                  => $candlepin_db_password,
-    db_ssl                       => $candlepin_db_ssl,
-    db_ssl_verify                => $candlepin_db_ssl_verify,
-    manage_db                    => $candlepin_manage_db,
-  } ~>
-  class { '::certs::qpid_client': } ~>
-  class { '::pulp':
-    oauth_enabled          => true,
-    oauth_key              => $katello::oauth_key,
-    oauth_secret           => $katello::oauth_secret,
-    messaging_url          => 'ssl://localhost:5671',
-    messaging_ca_cert      => $::certs::ca_cert,
-    messaging_client_cert  => $certs::qpid_client::messaging_client_cert,
-    messaging_transport    => 'qpid',
-    messaging_auth_enabled => false,
-    broker_url             => 'qpid://localhost:5671',
-    broker_use_ssl         => true,
-    consumers_crl          => $candlepin::crl_file,
-    proxy_url              => $proxy_url,
-    proxy_port             => $proxy_port,
-    proxy_username         => $proxy_username,
-    proxy_password         => $proxy_password,
-    yum_max_speed          => $pulp_max_speed,
-    manage_broker          => false,
-    manage_httpd           => false,
-    manage_plugins_httpd   => true,
-    manage_squid           => true,
-    enable_rpm             => true,
-    enable_puppet          => true,
-    enable_docker          => true,
-    enable_ostree          => $enable_ostree,
-    num_workers            => $num_pulp_workers,
-    max_tasks_per_child    => $max_tasks_per_pulp_worker,
-    enable_parent_node     => false,
-    repo_auth              => true,
-    puppet_wsgi_processes  => 1,
-    enable_katello         => true,
-  } ~>
-  class { '::qpid::client':
-    ssl                    => true,
-    ssl_cert_name          => 'broker',
-    ssl_cert_db            => $certs::nss_db_dir,
-    ssl_cert_password_file => $certs::qpid::nss_db_password_file,
-  } ~>
-  class { '::katello::qpid':
-    client_cert => $certs::qpid::client_cert,
-    client_key  => $certs::qpid::client_key,
+  $candlepin_url = "https://${candlepin_hostname}:8443/candlepin"
```

Comment:
I'm not sure what's best. I think for candlepin we could only expose the URL parameter rather than the hostname. @Klaas- what do you think?